src/hello_proto.h

```diff
@@ -19,15 +19,16 @@
 #define HELLO_INSTANCE_ID      0x0001
 #define HELLO_METHOD_ID        0x8001
 
-#define HELLO_EVENTGROUP_ID    0x0100
+#define HELLO_EVENTGROUP_ID    0x8005   // assume 1:1 mapping with event_id for eventgroup
```

Comment:
```suggestion
#define HELLO_EVENTGROUP_ID    0x8001   // assume 1:1 mapping with event_id for eventgroup
```

Reply:
Based on the COVESA Protos https://github.com/COVESA/uservices/blob/main/src/main/proto/example/hello_world/v1/hello_world_service.proto

Reply:
Though the default major version is defined as 1 in hellp_proto.h (#define HELLO_SERVICE_MAJOR    (int)(vsomeip::DEFAULT_MAJOR) // 1u) , client tried to connecting to service with default major as 0 

Hence shows in below log when client is started,
[HelloCli] [on_state] Requesting Hello Service [03e7.0000 v**0**.0] 
instead of 
[HelloCli] [on_state] Requesting Hello Service [03e7.0000 v**1**.0]

Reply:
> I suggest to update the values of service id, event id, event group id to be aligned as per COVESA protos
> 
> 
> 
> #define HELLO_SERVICE_ID       0x3E7
> 
> #define HELLO_EVENTGROUP_ID    0x8001   // assume 1:1 mapping with event_id for eventgroup
> 
> #define HELLO_EVENT_ID         0x8001
> 
> 
> 
> https://github.com/COVESA/uservices/blob/main/src/main/proto/example/hello_world/v1/hello_world_service.proto

You can change any ids, but better in separate pr

Reply:
> Though the default major version is defined as 1 in hello_proto.h (#define HELLO_SERVICE_MAJOR    (int)(vsomeip::DEFAULT_MAJOR) // 1u) , client tried to connecting to service with default major as 0 
> 
> 
> 
> Hence shows below in the logs when client application is started,
> 
> [HelloCli] [on_state] Requesting Hello Service [03e7.0000 v**0**.0] 
> 
> instead of 
> 
> [HelloCli] [on_state] Requesting Hello Service [03e7.0000 v**1**.0]
> 
> 

In previous release vsomeip had custom patch to use default major version 1. I reverted that as it is possible to use non default major services without patching vsomeip

Reply:
Yes that is correct. vsomeip applications can use non default major version. Where in hello-world project should I be mentioning the major version for the client to connect to ? 

(Example: I have a Hello world service started with major-1 and minor-0 via another tool
which file in hello world application should I mention these values to let the client connect to my service with major 1 and minor 0) ?

Reply:
or we should just align this application to use major minor versions also from COVESA protos, if we are planning to use this as base application for testing all the other tools ? @stevenhartley  https://github.com/COVESA/uservices/blob/main/src/main/proto/example/hello_world/v1/hello_world_service.proto

Reply:
Here is an extensive example how to change all the IDs and versions in docker, also possible in standalone versions:
https://github.com/int0x27/up-vsomeip-helloworld/tree/hello-service-customimzation?tab=readme-ov-file#docker-entrypoint-customization

Regarding major version difference, yes the change to previous version is actually in `vsomeip::DEFAULT_MAJOR`, before it was patched to 1, now it is left to default value: 0.
This is better as you may have unpatched vsomeip at the other end.

Reply:
Actually `hello_world_service.proto` seems to specify 2 different events:
- `0x8000` for one_second timer
- `0x8001` for one_minute timer
Am I correct @stevenhartley ? In that case the implementation needs to be reworked to send different event ids per timer and payload could be modified to exclude the timer id value.